staging/src/k8s.io/client-go/discovery/discovery_client_test.go

```diff
@@ -576,28 +581,64 @@ func TestGetOpenAPISchemaV3(t *testing.T) {
 		t.Fatalf("unexpected error getting openapi: %v", err)
 	}
 
-	for k, v := range paths {
-		actual, err := v.Schema()
-		if err != nil {
-			t.Fatal(err)
-		}
+	contentTypes := []string{
+		runtime.ContentTypeJSON, openapi.ContentTypeOpenAPIV3PB,
+	}
 
-		expected := testV3Specs[k]
-		if !golangproto.Equal(expected, actual) {
-			t.Fatalf("expected \n%v\n\ngot:\n%v", expected, actual)
-		}
+	for _, contentType := range contentTypes {
+		t.Run(contentType, func(t *testing.T) {
+			for k, v := range paths {
+				actual, err := v.Schema(contentType)
+				if err != nil {
+					t.Fatal(err)
+				}
 
-		// Ensure that fetching schema once again does not return same instance
-		actualAgain, err := v.Schema()
-		if err != nil {
-			t.Fatal(err)
-		}
+				expected := testV3Specs[k]
+				if contentType == runtime.ContentTypeJSON {
+					var actualSpec spec3.OpenAPI
 
-		if reflect.ValueOf(actual).Pointer() == reflect.ValueOf(actualAgain).Pointer() {
-			t.Fatal("expected schema not to be cached")
-		} else if !golangproto.Equal(actual, actualAgain) {
-			t.Fatal("expected schema values to be equal")
-		}
+					if err := json.Unmarshal(actual, &actualSpec); err != nil {
+						t.Fatal(err)
+					}
+
+					// Cannot use DeepEqual directly due to differences in how
+					// default key is being handled in gnostic vs kube-openapi
+					// Our test server parses the files in directly as gnostic
+					// which retains empty maps/lists, etc.
+					require.EqualValues(t, expected, &actualSpec)
+				} else {
```

Comment:
Maybe be slightly more defensive here if we add more types in the future?

Reply:
changed to switch which panics with unrecognized content type